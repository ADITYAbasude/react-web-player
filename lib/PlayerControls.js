"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const PlayIcon_1 = __importDefault(require("./icons/PlayIcon"));
const PauseIcon_1 = __importDefault(require("./icons/PauseIcon"));
const PlayerBottomContent_1 = __importDefault(require("./PlayerBottomContent"));
class PlayerControls extends react_1.Component {
    constructor(props) {
        super(props);
        this.onSeekingChange = (isSeeking) => {
            this.setState({ isSeeking });
        };
        this.onPlayPauseButtonClick = (isPlaying) => {
            this.setState({ isPlaying });
        };
        // handle play pause event
        this.handlePlayPauseButtonClick = () => {
            if (this.props.videoRef && this.props.videoRef.current) {
                const video = this.props.videoRef.current;
                if (video && video.paused) {
                    video.play();
                    this.setState({ isPlaying: true });
                }
                else {
                    video.pause();
                    this.setState({ isPlaying: false });
                }
            }
        };
        this.state = {
            isPlaying: false,
            isMuted: false,
            isLooped: false,
            isFullScreen: false,
            volume: 1,
            isSeeking: false,
        };
        this.handlePlayPauseButtonClick =
            this.handlePlayPauseButtonClick.bind(this);
    }
    // handle volume of video
    render() {
        const { isPlaying, isSeeking } = this.state;
        const { videoRef } = this.props;
        if (videoRef === null || videoRef === void 0 ? void 0 : videoRef.current) {
            videoRef.current.onended = () => {
                this.setState({ isPlaying: false });
            };
        }
        return (react_1.default.createElement("div", { className: `${isSeeking ? "wp-control-panel" : "wp-control-panel-hover"}` },
            react_1.default.createElement("button", { className: "wp-control-button wp-play-pause", "aria-label": "play/pause", onClick: this.handlePlayPauseButtonClick }, isPlaying ? react_1.default.createElement(PauseIcon_1.default, { size: "50" }) : react_1.default.createElement(PlayIcon_1.default, { size: "50" })),
            react_1.default.createElement(PlayerBottomContent_1.default, { onSeekingChange: this.onSeekingChange, onPlayPauseButtonClick: this.onPlayPauseButtonClick, videoRef: videoRef, isPlaying: isPlaying, onSeeking: this.props.onSeeking, color: this.props.color })));
    }
}
exports.default = PlayerControls;
