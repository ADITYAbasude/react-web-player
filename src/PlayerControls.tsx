import React, { Component, useState } from "react";
import PlayIcon from "./icons/PlayIcon";
import PauseIcon from "./icons/PauseIcon";
import PlayerBottomContent from "./PlayerBottomContent";

class PlayerControls extends Component<{
  videoRef?: React.RefObject<HTMLVideoElement>;
  onSeeking?: () => void;
  color?: string 
}> {
  constructor(props: any) {
    super(props);
    this.state = {
      isPlaying: false,
      isMuted: false,
      isLooped: false,
      isFullScreen: false,
      volume: 1,
      isSeeking: false,
    };
    this.handlePlayPauseButtonClick =
      this.handlePlayPauseButtonClick.bind(this);
  }

  onSeekingChange = (isSeeking: boolean) => {
    this.setState({ isSeeking });
  };

  onPlayPauseButtonClick = (isPlaying: boolean) => {
    this.setState({ isPlaying });
  };

  // handle play pause event
  handlePlayPauseButtonClick = () => {
    if (this.props.videoRef && this.props.videoRef.current) {
      const video = this.props.videoRef.current;
      if (video && video.paused) {
        video.play();
        this.setState({ isPlaying: true });
      } else {
        video.pause();
        this.setState({ isPlaying: false });
      }
    }
  };
  // handle volume of video

  render() {
    const { isPlaying, isSeeking } = this.state as {
      isPlaying: boolean;
      isSeeking: boolean;
    };
    const { videoRef } = this.props;

    if (videoRef?.current) {
      videoRef.current.onended = () => {
        this.setState({ isPlaying: false });
      }
    }

    return (
      <div
        className={`${
          isSeeking ? "wp-control-panel" : "wp-control-panel-hover"
        }`}
      >
        <button
          className="wp-control-button wp-play-pause"
          aria-label="play/pause"
          onClick={this.handlePlayPauseButtonClick}
        >
          {isPlaying ? <PauseIcon size="50" /> : <PlayIcon size="50" />}
        </button>

        <PlayerBottomContent
          onSeekingChange={this.onSeekingChange}
          onPlayPauseButtonClick={this.onPlayPauseButtonClick}
          videoRef={videoRef}
          isPlaying={isPlaying}
          onSeeking={this.props.onSeeking}
          color={this.props.color}
        />
      </div>
    );
  }
}

export default PlayerControls;
