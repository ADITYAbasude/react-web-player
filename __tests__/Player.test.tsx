import '@testing-library/jest-dom';
import React from "react";
import { render, screen } from "@testing-library/react";
import PlayerControls from "../src/PlayerControls";


test("render player component with play and pause button", () => {
  render(<PlayerControls />);
  const playButton = screen.getByRole('button', {name: /play/i});
  const pauseButton = screen.getByRole('button', {name: /pause/i});
  expect(playButton).toContainHTML('<svg');
  expect(pauseButton).toContainHTML('<svg');
});
