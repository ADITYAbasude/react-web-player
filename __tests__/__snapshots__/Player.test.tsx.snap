// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classname 1`] = `
<div>
  <div
    className="control-panel"
  >
    <button>
      <svg
        fill="currentColor"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M240,128a15.74,15.74,0,0,1-7.6,13.51L88.32,229.65a16,16,0,0,1-16.2.3A15.86,15.86,0,0,1,64,216.13V39.87a15.86,15.86,0,0,1,8.12-13.82,16,16,0,0,1,16.2.3L232.4,114.49A15.74,15.74,0,0,1,240,128Z"
        />
      </svg>
    </button>
    <button>
      <svg
        fill="currentColor"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M216,48V208a16,16,0,0,1-16,16H160a16,16,0,0,1-16-16V48a16,16,0,0,1,16-16h40A16,16,0,0,1,216,48ZM96,32H56A16,16,0,0,0,40,48V208a16,16,0,0,0,16,16H96a16,16,0,0,0,16-16V48A16,16,0,0,0,96,32Z"
        />
      </svg>
    </button>
    <button>
      Mute
    </button>
    <button>
      Volume
    </button>
    <button>
      FullScreen
    </button>
    <button>
      Loop
    </button>
  </div>
  <video
    src="https://www.youtube.com/watch?v=JGwWNGJdvx8"
    style={
      {
        "height": 500,
        "width": 500,
      }
    }
  />
</div>
`;
